src/controllers/productsImagesController.ts

```diff
@@ -0,0 +1,55 @@
+import { ProductsImagesModel } from './../models/productsImages';
+import { Request, Response } from "express";
+import cloudinary from "../config/fileSystem";
+
+export const createProductImages = async (req :Request,res:Response)=>{
+    try{
+        const productId = req.body.productId;
+        if(Number.isNaN(productId)){
+            return res.sendStatus(400);
+        }
+
+        const imageFile : any = req.files;
+        if(imageFile?.length == 0){
+            return res.status(400).json({error:"missing image"})
+        }
+        if( imageFile?.length > 10){
+            return res.status(400).json({error:"max allowed 10 images at once"})
+        }
+        const mainImages : any = [];
+        const expectedLength = imageFile.length;
+
+        
+        imageFile?.forEach(async(e : any)=>{
+                let base64Image = e.buffer.toString('base64');
+                let UploadedImage = await cloudinary.uploader.upload(`data:image/png;base64,${base64Image}`,{
```

Comment:
it is always better to wrap your 3rd party services in adapter pattern so in the future if you decided to change this service you can do it from your adapter instead of searching the project and changing it.

For example, you could have `imagesStore.uploadImage( .... )`